mlflow/sagemaker/__init__.py

```diff
@@ -772,11 +827,24 @@ def _update_sagemaker_endpoint(endpoint_name, image_url, model_s3_path, run_id,
                                 EndpointConfigName=new_config_name)
     eprint("Updated endpoint with new configuration!")
 
-    # If applicable, clean up unused models and old configurations
-    if not archive:
-        eprint("Cleaning up unused resources...")
+    def status_check_fn():
+        endpoint_info = sage_client.describe_endpoint(EndpointName=endpoint_name)
+        endpoint_update_was_rolled_back = (
+                endpoint_info["EndpointStatus"] == "InService"
```

Comment:
Is it guaranteed that after calling update_endpoint, the status immediately changes to InProgress? I'm worried about the case where there may be some delay, which could cause us to immediately consider the update failed.

Reply:
Added a mandatory 20 second delay before the endpoint status is used to determine whether or not the update succeeded